Sources/UI/Chat/Chat View Controller/ChatViewController+MessageActions.swift

```diff
@@ -297,3 +298,136 @@ extension ChatViewController {
             .disposed(by: disposeBag)
     }
 }
+
+// MARK: - Context Menu
+
+@available(iOS 13, *)
+extension ChatViewController {
+    
+    public func tableView(_ tableView: UITableView,
+                          contextMenuConfigurationForRowAt indexPath: IndexPath,
+                          point: CGPoint) -> UIContextMenuConfiguration? {
+        guard useContextMenuForActions else {
+            return nil
+        }
+        
+        return UIContextMenuConfiguration(identifier: nil, previewProvider: nil) { [weak self] _ in
+            guard let self = self,
+                let cell = tableView.cellForRow(at: indexPath),
+                let message = self.channelPresenter?.items[indexPath.row].message else {
+                return nil
+            }
+            
+            let locationInView = tableView.convert(point, to: cell)
+            return self.createActionsContextMenu(from: cell, for: message, locationInView: locationInView)
+        }
+    }
+    
+    public func defaultActionsContextMenu(from cell: UITableViewCell, for message: Message, locationInView: CGPoint) -> UIMenu? {
```

Comment:
Why do we need this function? Why can't this code be in `createActionsContextMenu`?

Reply:
It's a lot of code there and it's better to move it in a separate extension.

Reply:
Do we need to have it in `ChatViewController.swift`? Can't we put `createActionsContextMenu` here?